lib/soapy_bing/ads/campaign_performance_report.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'soapy_bing/ads/report'
+require 'date'
+
+module SoapyBing
+  class Ads
+    class CampaignPerformanceReport < Report
+      attr_reader :date_range
+
+      def initialize(options)
+        super
+        @date_range = Range.new(
+          Date.parse(options.fetch(:date_start)),
+          Date.parse(options.fetch(:date_end))
```

Comment:
It seems a bit awkward to me that start/end dates are passed through 2 functions down the stack, are merged into one object here, and then are still used separately in this same class as `date_range.start` and `date_range.end`. I would either pass them separately all the way, or pass them as a range and decompose in the place of actual use.